build/scripts/check-msbuild.ps1

```diff
@@ -1,18 +1,28 @@
 # Examine an MSBuild log and issue an error if any MSBulid warnings are discovered.
 
 param ([string]$logFile = $(throw "Need a build log file"))
+set-strictmode -version 2.0
 $ErrorActionPreference="Stop"
 
 try
 {
     $exitCode = 0
     $pattern = "\bwarning\b\s+\bMSB(\d+):(.*)"
+    $ignoreCase = [System.Text.RegularExpressions.RegexOptions]::IgnoreCase
     foreach ($line in gc $logFile) {
-        $m = [regex]::match($line, $pattern, [System.Text.RegularExpressions.RegexOptions]::IgnoreCase)
+        $m = [regex]::Match($line, $pattern, $ignoreCase)
         if ($m.Success) {
             $num = $m.Groups[1].Value
             $text = $m.Groups[2].Value
             $location = $line.Substring(0, $m.Index)
+
+            # MSBuild race condition.  Doesn't break build so don't promote to error for now
+            #
+            # https://github.com/dotnet/roslyn/issues/10116
```

Comment:
Both.  It's basically tracking properly dealing with this "hack" I'm about to check in.